packages/eslint-plugin/src/rules/no-empty-styled-expression/__tests__/rule.test.ts

```diff
@@ -0,0 +1,75 @@
+import type { RuleTester } from 'eslint';
+
+import { typeScriptTester as tester } from '../../../test-utils';
+import { noEmptyStyledExpressionRule } from '../index';
+
+type InvalidTestCase = Omit<RuleTester.InvalidTestCase, 'errors'>;
+
+const createInvalidTestCases = (tests: InvalidTestCase[]) =>
+  tests.map((t) => ({
+    ...t,
+    errors: [{ messageId: 'unexpected' }],
+  }));
+
+tester.run('no-styled-tagged-template-expression', noEmptyStyledExpressionRule, {
+  valid: [
+    `
+        import { styled } from 'styled';
+
+        styled.div({
+            color: blue
+        });
+    `,
+    `
+        import { styled } from 'styled';
+
+        styled.span({
+            color: blue
+        });
+    `,
+    `
+        import { styled } from '@compiled/react';
+
+        styled.span("hello world");
```

Comment:
Not sure if styled.span("Hello world") is valid Compiled syntax, will need to double check

Can we also add some examples of using a template string too?

```js
styled.span`color: blue`;

styled.h2`
    color: blue;
    background-color: pink;
`;
```

This was my mistake (sorry!) for not mentioning it in the ticket, but this is a loophole that you may want to fix with this rule as well:

```js
styled.div``; // equivalent to styled.div({})
styled.span``; // equivalent to styled.span({})
```

Reply:
Isn't the template usage disallowed in the first place? Covered [by this rule](https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin/src/rules/no-styled-tagged-template-expression). It doesn't check for empty template expressions, but it _**changes template expressions => CallExpressions**_ where this no-empty-syled-expression rule will then apply. It seems a little redundant to check this case, but please correct me if I'm wrong!

Reply:
Discussed with Tom and Liam -- as the aforementioned syntax is covered by [another ESLint rule](https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin/src/rules/no-styled-tagged-template-expression), checks for the TagTemplateExpression will not be added to this PR.

Reply:
ah sounds good, ty for the extra context!